arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/stream/concurrent/PubSub.kt

```diff
@@ -659,13 +658,13 @@ internal class DefaultPubSub<I, O, QS, S>(private val strategy: PubSub.Strategy<
    */
   private tailrec fun loop(
     ps: PubSub.PubSubState<I, O, QS, S>,
-    action: suspend () -> Unit
-  ): Pair<PubSub.PubSubState<I, O, QS, S>, (suspend () -> Unit)> {
```

Comment:
The actions running in the `loop` no longer need to be suspending since we rely on `UnsafePromise` for signaling.

Instead the functionality of `suspend` is exposed through the `modify` operator.